src/appleseed/renderer/kernel/shading/closures.cpp

```diff
@@ -2462,10 +2463,11 @@ bool process_matte_tree(
                 const OSL::ClosureAdd* c = reinterpret_cast<const OSL::ClosureAdd*>(closure);
                 foundation::Color3f color_a(0.0f), color_b(0.0f);
                 Alpha alpha_a(0.0f), alpha_b(0.0f);
-                process_matte_tree(c->closureA, color_a, alpha_a);
-                process_matte_tree(c->closureB, color_b, alpha_b);
+                bool is_matte_a = process_matte_tree(c->closureA, color_a, alpha_a);
+                bool is_matte_b = process_matte_tree(c->closureB, color_b, alpha_b);
                 matte_color = color_a + color_b;
                 matte_alpha.set(alpha_a[0] + alpha_b[0]);
+                return is_matte_a || is_matte_b;
```

Comment:
I made this part `return process_matte_tree(c->closureA, color_a, alpha_a) || process_matte_tree(c->closureB, color_b, alpha_b)`, as you suggested but then I thought maybe we need to combine matte colors? I'm not sure if it's needed, but it's possible to blend mattes with blend shader.